### ISSUE TITLE
stack-1.6.3 needs unliftio-0.2.1.0

### ISSUE THREAD

**Issue by kirelagin** [NixOS/nixpkgs#33097]
The [build of stack-1.6.3](https://hydra.nixos.org/build/66599924) fails because of [this commit](https://github.com/commercialhaskell/stack/commit/b662e9b452c5f8f4a5f720b1d41c1c836d8a67fe). Many IO-related functions (`hSetBuffering` is one of them) now come from `UnliftIO.IO`, where they appeared in 0.2.1.0, instead of `System.IO`.

I am not sure whether upgrading `unliftio` or downgrading `stack` is a better solution, but it would definitely be nice to have working `stack`, one way or another 🤔. cc @peti

**Comment by chris-martin** [comment]
Here's the override that needs to be applied:

        unliftio = callPackage
          ({ mkDerivation, async, base, deepseq, directory, filepath
           , transformers, unix, unliftio-core
           }:
           mkDerivation {
             pname = "unliftio";
             version = "0.2.1.0";
             sha256 = "09i17ajszvr1cm4x31k157mlfinxsmrmkxmjsmlvsxvch58m2lgi";
             libraryHaskellDepends = [
               async base deepseq directory filepath transformers unix
               unliftio-core
             ];
             homepage = "https://github.com/fpco/unliftio/tree/master/unliftio#readme";
             description = "The MonadUnliftIO typeclass for unlifting monads to IO (batteries included)";
             license = stdenv.lib.licenses.mit;
           }) {};

But I haven't been able to figure out where to put it.

**Comment by chris-martin** [comment]
How did this happen? Did Stack really publish a revision of [all-cabal-hashes](https://github.com/commercialhaskell/all-cabal-hashes/commit/72f75113ede74a720c73526754eeb8b0dc4f5f9a) in which Stack doesn't compile?

**Comment by kirelagin** [comment]
> Did Stack really publish a revision of all-cabal-hashes in which Stack doesn't compile?

What do you mean? As far as I understand it, `all-cabal-hashes` simply contains what it says – hashes of everything. And the version you pointed to contains `unliftio-0.2.1.0` alongside other versions of it.

I have no clue how `hackage-packages.nix` is generated (it would be helpful to have this explained in the headers of all auto-generated files 😊) but the point is that `unliftio-0.2.0.0` ended up there together with `stack-1.6.3` which has a lower bound of `0.2.1.0` on it.

**Comment by peti** [comment]
The version of `stack` was updated faster than the version of `unliftio` because the latter is guided by the LTS-10.x package set, which publishes updates only once per week, whereas updates to `stack` are taken directly from Hackage and make it into Nixpkgs more or less immediately. Normally, the update to stack would have been delayed to keep it building, but it appears that the build failure flew by under the radar, i.e. I just didn't notice it on Hydra before merging the update. 415bda6c8f4091c0894d922f442f19f31877bc20 will fix this issue once it lands on `master` (within a day or two at most).
